c++/src/kj/compat/http.h

```diff
@@ -492,6 +494,27 @@ class HttpHeaders {
   //   also add direct accessors for those headers.
 };
 
+struct HttpByteRange {
+  // Inclusive HTTP range
+
+  uint64_t start;
+  uint64_t end;
+
+  inline bool operator==(const HttpByteRange& other) const {
+    return start == other.start && end == other.end;
+  }
+  inline bool operator!=(const HttpByteRange& other) const {
+    return !(start == other.start && end == other.end);
+  }
+};
+
+kj::String KJ_STRINGIFY(HttpByteRange range);
+
+kj::Maybe<kj::Array<HttpByteRange>> tryParseHttpRangeHeader(kj::ArrayPtr<const char> value, uint64_t contentLength);
```

Comment:
(Non-actionable musings.)

It's a bit funny that we can't actually parse this header until we know the size of the resource. That implies you can't fully validate the headers until after looking up the file and querying it. Theoretically the API should be something like... parse into a representation of the header first, then validate that representation against the actual content length in a separate step, producing the final set of ranges.

But given this works fine for the present use case I guess it's fine. If we introduce a more complicated API later we can always implement this version in terms of it.